[docs] runtime version doc page (#13819)
expo/expo: docs/pages/distribution/runtime-version.md

---

> title: Runtime Version

Reviewer: What is a runtime version?
Do I need to set runtime version in my project?
When should I change my runtime version?

Reviewer 2: ```suggestion
title: Runtime Versions
```
(also rename the file to runtime-versions.md)

---

> 

Reviewer 3: ```suggestion

> Custom runtime versions are not supported on the classic build system (`expo build`); these apps will always use the SDK version as the basis for determining runtime compatibility. 

```

> Expo Update allows you to publish over the air updates to apps. The update must be for an app running a

Reviewer 2: ```suggestion
Expo Updates allow you to publish over-the-air updates to apps. The update must be for an app running a
```

> [compatible](../workflow/publishing/#what-version-of-the-app-will-my) runtime. You may use a runtime other than an Expo SDK, by specifying a runtime version.

Reviewer: Are we updating this section too?

Author: yea once the "main" description stabilizes I will go and look at all the references.

> Expo Update allows you to publish over the air updates to apps. The update must be for an app running a
> [compatible](../workflow/publishing/#what-version-of-the-app-will-my) runtime. You may use a runtime other than an Expo SDK, by specifying a runtime version.

Reviewer 3: ```suggestion
Over-the-air updates with `expo-updates` will only work in binaries with a [compatible](../workflow/publishing/#what-version-of-the-app-will-my) runtime. By default, the runtime version is the Expo SDK version: the runtime is the set of libraries in the Expo SDK. Apps built with EAS Build will include only the dependencies that are present at the time that the app binary is built, and so the Expo SDK version will often not properly describe the runtime of the app. In these cases, you can start using `runtimeVersion` and it will take precedence over the Expo SDK version.
```

The runtime version should be specified in your `app.json`:

> ```typescript

Reviewer 3: ```suggestion
```json
```

{

> 	expo: {

Reviewer 3: two spaces indentation. use valid json

		...

> 		runtimeVersion: "2.718",

Reviewer: What is a valid runtime version? Does it have to be semver?

	}
}
```

> ## Publishing

Reviewer 3: ```suggestion
## Setting the runtime version for an update
```

> [Publishes](../workflow/publishing/#how-to-publish) run with the runtime version set in the `app.json` will be delivered to builds running the same runtime version.

Reviewer 3: ```suggestion
[Updates](/workflow/publishing.md#how-to-publish) run with the runtime version set in the `app.json` will be delivered to builds running the same runtime version.
```

> ## Builds

Reviewer 3: ```suggestion
## Setting the runtime version for a build
```

> Apps running a custom runtime version can be built with [EAS Build](../build/introduction/).

Reviewer 3: ```suggestion
Apps running a custom runtime version can be built with [EAS Build](/build/introduction.md).
```

> There are two ways to set the runtime version of a build.

Reviewer 3: these should be split up based on whether you have a managed app or not. if you have a managed app then you should just update your app config (app.json/app.config.js), otherwise you should edit native files directly. we should leave out `expo prebuild`

1. (Recommended) After setting the runtime version in your `app.json`, run `expo prebuild`.

> 2. Manually edit the Plist/AndroidManifest.xml:

Reviewer: Info.plist / expo.plist ?

Reviewer 2: ```suggestion
2. Edit Expo.plist on iOS and AndroidManifest.xml on Android. In Expo.plist, add an entry whose key is `EXUpdatesRuntimeVersion` and value is a string set to the desired runtime version. In AndroidManifest.xml, add a `<meta-data>` element whose `android:name` attribute is `expo.modules.updates.EXPO_RUNTIME_VERSION` and `android:value` attribute is the desired runtime version.
```

	### iOS

	```diff
	+ <key>EXUpdatesRuntimeVersion</key>

>     + <string>2.718</string>

Reviewer 3: indentation looks off?

Reviewer 2: This might be glitchy behavior with GHM since the preview looks OK but it would be nice for the source to look well-formatted if possible.

Author: Oh weird, this is. a tab/space issue that rendered as equal on my local setup. Fixed.
